File: src/services/conversation_service.py
"""Conversation management service"""
from datetime import datetime
from typing import List, Dict
from src.config.openai_config import SYSTEM_MESSAGE
from src.database.manager import DatabaseManager

class ConversationService:
    def __init__(self):
        self.db = DatabaseManager()
        self.current_conversation_id = None
        self.messages = [
            {"role": "system", "content": SYSTEM_MESSAGE}
        ]
    
    def start_new_conversation(self) -> int:
        """Start a new conversation"""
        title = f"Chat {datetime.now().strftime('%Y-%m-%d %H:%M')}"
        self.current_conversation_id = self.db.create_conversation(title)
        self.messages = [{"role": "system", "content": SYSTEM_MESSAGE}]
        return self.current_conversation_id
    
    def add_message(self, role: str, content: str) -> None:
        """Add a message to the current conversation"""
        if not self.current_conversation_id:
            self.start_new_conversation()
        
        message = {"role": role, "content": content}
        self.messages.append(message)
        self.db.add_message(self.current_conversation_id, role, content)
    
    def get_messages(self) -> List[Dict[str, str]]:
        """Get all messages in the current conversation"""
        return self.messages
    
    def clear_conversation(self) -> None:
        """Clear the current conversation"""
        self.current_conversation_id = None
        self.messages = [{"role": "system", "content": SYSTEM_MESSAGE}]

File: src/services/ConversationService.js
import { DatabaseManager } from '../utils/database/DatabaseManager.js';

export class ConversationService {
  constructor() {
    this.db = new DatabaseManager();
    this.currentConversationId = null;
  }

  async getConversations() {
    return this.db.getConversations();
  }

  async startNewConversation() {
    const title = `Chat ${new Date().toLocaleString()}`;
    this.currentConversationId = await this.db.createConversation(title);
    return this.currentConversationId;
  }

  async addMessage(role, content) {
    if (!this.currentConversationId) {
      await this.startNewConversation();
    }
    return this.db.addMessage(this.currentConversationId, role, content);
  }

  async loadConversation(conversationId) {
    this.currentConversationId = conversationId;
    return this.db.getConversationMessages(conversationId);
  }

  async deleteConversation(conversationId) {
    await this.db.deleteConversation(conversationId);
    if (conversationId === this.currentConversationId) {
      this.currentConversationId = null;
    }
  }
}